test/subscriber.ts

```diff
@@ -290,6 +291,68 @@ describe('Subscriber', () => {
       assert.strictEqual(subscriber.ackDeadline, fakeDeadline);
     });
 
+    it('should bound ack deadlines if min/max are specified', () => {
+      const histogram: FakeHistogram = stubs.get('histogram');
+      const now = Date.now();
+
+      message.received = 23842328;
+      sandbox.stub(global.Date, 'now').returns(now);
+
+      const expectedSeconds = (now - message.received) / 1000;
+      const addStub = sandbox.stub(histogram, 'add').withArgs(expectedSeconds);
+
+      let fakeDeadline = 312123;
+      sandbox
+        .stub(histogram, 'percentile')
+        .withArgs(99)
+        .callsFake(() => fakeDeadline);
+
+      subscriber.setOptions({
+        maxAckDeadline: Duration.from({seconds: 60}),
+      });
+      subscriber.ack(message);
+
+      assert.strictEqual(addStub.callCount, 1);
+      assert.strictEqual(subscriber.ackDeadline, 60);
+
+      subscriber.setOptions({
+        minAckDeadline: Duration.from({seconds: 10}),
+      });
+      fakeDeadline = 1;
+      subscriber.ack(message);
+
+      assert.strictEqual(subscriber.ackDeadline, 10);
+    });
```

Comment:
What 'time' does Duration.from actually affect? Is it the moment subscriber.ack is called? Feel free to ignore if this doesn't make sense, but it would be nice to have a test where the deadline matches your expectations of the actual time of when it should be called, for example, exactly a minute later, etc.